src/plugins/pagination.js

```diff
@@ -174,7 +175,7 @@ module.exports = function paginationPlugin (bookshelf) {
         });
         qb.countDistinct.apply(qb, [`${tableName}.${idAttribute}`]);
 
-      }).fetchAll().then(result => {
+      }).fetchAll({transacting}).then(result => {
```

Comment:
would it make more sense to pass in fetchOptions directly here?
I'm using the bookshelf-paranoia plugin and it passes in withDeleted option,  which pulls in records if they're softdeleted or not.  Since it doesn't get passed to the pagination plugin, it only counts rows that are not soft deleted.